{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Breast Cancer Proteome\n",
    "\n",
    "Data\n",
    "https://www.kaggle.com/piotrgrabo/breastcancerproteomes\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Description\n",
    "\n",
    "### Context: \n",
    "This data set contains published iTRAQ proteome profiling of 77 breast cancer samples generated by the Clinical Proteomic Tumor Analysis Consortium (NCI/NIH). It contains expression values for ~12.000 proteins for each sample, with missing values present when a given protein could not be quantified in a given sample.\n",
    "\n",
    "### Content:\n",
    "\n",
    "#### File: 77_cancer_proteomes_CPTAC_itraq.csv\n",
    "\n",
    "- RefSeq_accession_number: RefSeq protein ID (each protein has a unique ID in a RefSeq database)\n",
    "- gene_symbol: a symbol unique to each gene (every protein is encoded by some gene)\n",
    "- gene_name: a full name of that gene \n",
    "- Remaining columns:\n",
    "    - each column is a person indicated by ID number\n",
    "    - __log2 iTRAQ ratios for each sample__ (protein expression data, most important!), three last columns are from healthy individuals\n",
    "\n",
    "#### File: clinical_data_breast_cancer.csv\n",
    "\n",
    "First column \"Complete TCGA ID\" lists 105 people, (note 28 were dropped)\n",
    "     - use to match the sample IDs in the cancer proteomes file (see example script). \n",
    "All other columns have self-explanatory names, contain data about the cancer classification of a given sample using different methods. 'PAM50 mRNA' classification is being used in the example script.\n",
    "\n",
    "#### File: PAM50_proteins.csv\n",
    "\n",
    "Contains the list of 100 genes / proteins used by the PAM50 classification system. \n",
    "- The column RefSeqProteinID contains the protein IDs that can be matched with the IDs in the cancer proteomes data set.\n",
    "\n",
    "### Past Research: \n",
    "The original study: http://www.nature.com/nature/journal/v534/n7605/full/nature18003.html (paywall warning)\n",
    "\n",
    "In brief: the data were used to assess how the mutations in the DNA are affecting the protein expression landscape in breast cancer. Genes in our DNA are first transcribed into RNA molecules which then are translated into proteins. Changing the information content of DNA has impact on the behavior of the proteome, which is the main functional unit of cells, taking care of cell division, DNA repair, enzymatic reactions and signaling etc. They performed K-means clustering on the protein data to divide the breast cancer patients into sub-types, each having unique protein expression signature. They found that the best clustering was achieved using 3 clusters (original PAM50 gene set yields four different subtypes using RNA data).\n",
    "\n",
    "### Inspiration:\n",
    "\n",
    "This is an interesting study and I myself wanted to use this breast cancer proteome data set for other types of analyses using machine learning that I am performing as a part of my PhD. However, I though that the Kaggle community (or at least that part with biomedical interests) would enjoy playing with it. I added a simple K-means clustering example for that data with some comments, the same approach as used in the original paper. One thing is that there is a panel of genes, the PAM50 which is used to classify breast cancers into subtypes. This panel was originally based on the RNA expression data which is (in my opinion) not as robust as the measurement of mRNA's final product, the protein. Perhaps using this data set, someone could find a different set of proteins (they all have unique NP_/XP_ identifiers) that would divide the data set even more robustly? Perhaps into a higher numbers of clusters with very distinct protein expression signatures?\n",
    "\n",
    "### Example K-means analysis script: \n",
    "http://pastebin.com/A0Wj41DP\n",
    "- uses K-means cluster (kajot, Berlin)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "My Notes\n",
    "- This data was generated  by the Clinical Proteomic Tumor Analysis Consortium (CPTAC) (NCI/NIH)\n",
    "- N=77 patients\n",
    "    - this is a very small sample size\n",
    "    - with flu I had 45 and could just about tell the difference between 15 patients and 30 patients with ~7 factors\n",
    "    - rule of thumb, 10-fold more samples than factors, so only use 7 factors if analyzing this data\n",
    "    - 100 proteins detected by iTRAQ proteome profiling\n",
    "    - 12,500 proteins detected with relative expression values for each sample\n",
    "        - when a protein was not detected there will be NaN\n",
    "        - what are these values normalized to? how are there negatives?\n",
    "    \n",
    "- What is PAM50 classification? \n",
    "    - there are Basal-like, HER2-enriched, Luminal A, & Luminal B classes\n",
    "    - PAM50 only uses 100 proteins\n",
    "    - PAM50 was based on mRNA\n",
    "    \n",
    "    \n",
    "- I should run an unsupervised clustering on this before I look at the types of proteins that were detected\n",
    "\n",
    "\n",
    "- What are the following columns\n",
    "    - Converted stage\n",
    "        - \n",
    "    - ER status*\n",
    "        - estrogen receptor\n",
    "    - PR status*\n",
    "        - progesterone receptor\n",
    "    - HER2*\n",
    "        - A central goal in breast cancer research has been the identification of druggable kinases beyond HER2\n",
    "        - \n",
    "    - Tumor\n",
    "        - The variable Tumor tells something about the size of the tumor and if the tumor is growing into the chest wall or skin. For 15 individuals their breast tumor was at stage T1 when diagnosed. For 65 individuals their breast tumor was at stage T2 when diagnosed. For 19 individuals their breast tumor was at stage T3 when diagnosed. For 6 individuals their breast tumor was at stage T4 when diagnosed\n",
    "    - Tumor: T other vs T1\n",
    "    - Node: 0-3 & pos/neg\n",
    "        - The variable Node indicates the degree of spread to regional lymph nodes. For 53 individuals tumor cells are absent from regional lymph nodes. In 29 individuals there is regional lymph node metastatis present. In 14 individuals there are metastasis present in 4 to 9 lymph nodes. And in 9 individuals metastatasis is present in more than 9 lymph nodes\n",
    "    - AJCC Stage & converted stage\n",
    "        - Staging determines whether the cancer has spread & if so how far\n",
    "        - this is determined by using Tumor, node, & metastasis\n",
    "        - more information: http://www.cancer.org/cancer/breastcancer/detailedguide/breast-cancer-staging\n",
    "    - OS event & Time\n",
    "    - RPPA clusters*\n",
    "    - CN clusters\n",
    "    \n",
    "- Delete patients  \n",
    "    - Take out the two people with metastasis there are not enough to do anything with\n",
    "    - \n",
    "    \n",
    "- What can I do with age? \n",
    "    - what is the spread of age?\n",
    "    - drop a few patients who are outside of a 10-20 year range?\n",
    "\n",
    "- Is there a reason to drop deceased (i.e. cancer was further progressed when sample was taken?)\n",
    "\n",
    "- Ignore columns\n",
    "    - Days to Date of Last Contact because it does not give health related info (individuals seek medical help at different stages, stage at time of sampling is more important)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Questions to answer using the data\n",
    "- which features (proteins) drive any found classification? (this is useful because it can elucidate potential therapeutical targets or shed light into novel cancer biology)\n",
    "- Which features are at the root of correlation\n",
    "    - perform Prinicpal Component Analysis (PCA) prior to performing hierarchical clustering in order to reduce the dimensionality of our data\n",
    "    - requires no missing values\n",
    "    - omit patients with missing values or proteins with missing values?\n",
    "        - look at distribution of missing values\n",
    "            - are there patients with loads missing or proteins with loads missing?\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Kernals with good lessons\n",
    "\n",
    "- Hierarchical Clustering (Anthony Leo, Detriot, Software engineer @ Plex Systems)\n",
    "    - https://www.kaggle.com/anthonyleo/hierarchical-clustering\n",
    "    - objective: to analyze the proteom data set to determine if there is any underlying structure and practice implementing Principal Component Analysis and Hierarchical Clustering\n",
    "    - use the R prcomp function to perform Prinicpal Component Analysis\n",
    "    - determine the number of principal components to use\n",
    "        - Each principal component captures a percentage of the variance within our data, and we can see exactly how much variance each principal component accounts for by creating a scree plot\n",
    "        - 8 principal components (PC) account for 52.6% of the variance, which is rather low\n",
    "    - pass the principal component vectors into the hclust function\n",
    "    - focus on 4 clusters\n",
    "    - plot PC1 v PC2\n",
    "    - Conclude: It appears that there is not a lot of underlying structure to this dataset. Since our principal component analysis was not able to effectively reduce the dimensionality of our data, we were not able to create well defined clusters using hierarchical clustering, and it is likely that most clustering methods would also fail on this dataset due to the curse of dimensionality.\n",
    "    - other pages by Anthony Leo\n",
    "        - https://www.kaggle.com/anthonyleo/linear-regression\n",
    "        - https://www.kaggle.com/anthonyleo/hypothesis-testing\n",
    "\n",
    "- Breast Cancer Sub-Characterization w/ Genomic Data (Jason T. Smith, Antarctica)\n",
    "    - https://www.kaggle.com/jasontsmith2718/breast-cancer-sub-characterization-w-genomic-data\n",
    "    - Linear Discriminant Analysis (LDA) was used to reduce dimensionality\n",
    "    - graph dimension x by dimension y (similar to Heirarchical clustering above)\n",
    "    - decision tree & x-fold cross validation\n",
    "    - ROC curves: A receiver operating characteristic curve, i.e. ROC curve, is a graphical plot that illustrates the diagnostic ability of a binary classifier system as its discrimination threshold is varied\n",
    "    - higher Area Under an ROC curve = better\n",
    "    - what is a multi-class AUROC metric?\n",
    "        - see sklearn example at 'http://scikit-learn.org/stable/auto_examples/model_selection/plot_roc.html'\n",
    "\n",
    "- Data Visualization of Molecular Subtype (Jason Huang, Data science student, Penn)\n",
    "    - https://www.kaggle.com/jwlzdh1/data-visualization-of-molecular-subtype\n",
    "    - This guy even has intro slides of prevalence (very impressive!)\n",
    "    - It is common that genetic datasets exhibit cases where n, the number of observations is larger than the p, the number or predictors. When p >n, there is no longer a unique least squares coefficient estimate. Often when p>n, high variance and overfitting are a major concern in this setting. Thus, simple, highly regularized approaches often become the methods of choice. To address this issue, there are many techniques (__Subset selection), (Lasso and Ridge) and (Dimension reduction__) to exclude irrelevant variables from a regression or a dataset.\n",
    "    - Use Lasso (R) to shrink dataset from 12553 predictors to 57 relevant variables\n",
    "    - Look for correlations between the 57 protein variables and specific tumor types (but what if tumor types aren't good classifications to determine root cause of disease or cures?)\n",
    "    - Plot variable importance to identify the 5 most \"influencial\" genes that will be used for visualization\n",
    "    - Graphs\n",
    "        - Pie chart to show 4 tumor types: Luminal A/B, Basal, HER2\n",
    "        - ...\n",
    "        - Basal can be differentiated from the other 3 by Mical & HNF3g (hepatocyte nuclear factor)\n",
    "            - box plot & violin plot of mical by the 4 cancer types\n",
    "            \n",
    "            \n",
    "- Ensemble Clustering (Marco Pietrosanto, Italy)\n",
    "    - https://www.kaggle.com/noise42/ensemble-clustering\n",
    "    - good at inspecting information in spreadsheets\n",
    "    - create random set of proteins\n",
    "    - nice heatmap\n",
    "    - ensemble clustering\n",
    "    - import itertools\n",
    "    - from sklearn.cluster import DBSCAN\n",
    "    - visualize with K==3\n",
    "    - construct a cooccurrence (consensus) matrix \n",
    "        - sns.clustermap(MatrixData)\n",
    "    - from scipy.cluster.hierarchy import dendrogram\n",
    "    - used pre-defined categories of tumors\n",
    "    \n",
    "    \n",
    "- IntroUnsupervisedAnalysis (srhoades10, Pharmacology PhD, Penn)\n",
    "    - https://www.kaggle.com/srhoades10/introunsupervisedanalysis\n",
    "    - histogram graph of protein intensity distribution\n",
    "    - 3D graph of principal components\n",
    "    - #Plot the clusters and the observations with respect to the cluster boundaries\n",
    "    - #Some of these commands are adapted from the scikit-learn example for KMeans\n",
    "    - #... which is found at http://scikit-learn.org/stable/auto_examples/cluster/plot_kmeans_digits.html#sphx-glr-auto-examples-cluster-plot-kmeans-digits-py (thanks!)\n",
    "    - 5D graph of principal components\n",
    "    - very few comments, no analysis or conclusion\n",
    "\n",
    "\n",
    "- Breast cancer markers using simple tools ( Efejiro Ashano, Bioinformatics @ NIBDA, Lagos Nigeria)\n",
    "    - aim to find early markers of breast cancer in the data \n",
    "    - compares to the three healthy people, which is no good, because there are only 3 of them! (I will try to differentiate between types of cancers)\n",
    "    - box plots\n",
    "    \n",
    "    \n",
    "- Clustering and analysis on Breast Cancer (ML_Enthusiast)\n",
    "    - graphs miRNA vs methylation clustering for different stages\n",
    "\n",
    "- Patient characteristics (Anika)\n",
    "    - https://www.kaggle.com/sinaasappel/patient-characteristics\n",
    "    - good analysis!\n",
    "    - delete the males!!!\n",
    "    - focus on age 40-70\n",
    "    - focus on tumor T1 & T2, smaller tumors (<2-5cm)\n",
    "    - focus on node N0, N1 (1-3), & N2(4-9) (spread to 0-9 lymphnodes)\n",
    "    - ignore staging because it is determined using tumor & node\n",
    "    \n",
    "    \n",
    "- Clustering proteins (Pete Bleackley)\n",
    "    - https://www.kaggle.com/petebleackley/clustering-proteins\n",
    "    - good list of sklearn programs to import\n",
    "    - identifies 8 protein clusters\n",
    "    - def ReduceDimensions(data,keep_variance=0.8,tolerance=1.0E-12,max_iter=1024)\n",
    "    - Previous work on this dataset has involved clustering patients according to their protein activity. For this analysis, I decided to cluster proteins according to their activity in different patients, and then use the activity of the different clusters to predict clinical features.\n",
    "        - predict Tumor and Node by proteins!\n",
    "    - First, the dimensionality of dataset is reduced, and then a hierarchical cluster is fitted to the principal components, which is visualised as a dendrogram to select the appropriate number of components\n",
    "\n",
    "    \n",
    "-  Cancer Proteomes vs Clinical Data Analysis (mbschultz1)\n",
    "    - https://www.kaggle.com/mbschultz1/cancer-proteomes-vs-clinical-data-analysis\n",
    "    - similar analysis to Anika (above)\n",
    "    - the rest is bad\n",
    "    \n",
    "\n",
    "-  Kmeans-3d-plot (AkhilPrakash)\n",
    "    - https://www.kaggle.com/prakashakhil/kmeans-3d-plot\n",
    "    - a 3D plot\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Plan of attack\n",
    "\n",
    "Question: \n",
    "\n",
    "Do clusters determined by using all proteins present in larger & further spread tumors lead to good definition of clusters for smaller & less spread tumors?  \n",
    "- Then can we compare expression levels of each protein expressed in low-T/N tumors to expression levels in high-T/N tumors to show which kinases will become more over expressed?\n",
    "    - compute change in expression levels between low & high-TN tumors of same cluster, \n",
    "    - sort proteins by amount of change\n",
    "    - greater change means greater potential for increased expression and therefore makes those proteins more likely to be good drug targets \n",
    "    - make a list of top 10 targets\n",
    "\n",
    "Can clustering by ALL proteins predict T & N of patients better than clustering by the 35 PAM50 proteins? \n",
    "\n",
    "\n",
    "(1) Drop NaNs\n",
    "- drop males\n",
    "- drop healthy individuals\n",
    "- drop pateints that are too old or young, graph, but probably focus on ages 40-70  \n",
    "- Look at proteins and drop proteins that have a large number of missing values\n",
    "- Look at samples and drop samples that have a large number of missing values\n",
    "- Look at distribution of missing values and drop more proteins and/or samples (or fill these in with average inferred values)\n",
    "\n",
    "\n",
    "(2) Use correlation between proteins expressed at later stage of disease (characterized by N, O) and treatment groups (ER, PR, HER2) to predict correlation between earlier disease and treatment group\n",
    "- maybe: Drop top 8-ish proteins with most variance - use data to predict which one will be \n",
    "\n",
    "\n",
    "(3) reduce dimensionality\n",
    "- \"dimension reduction\"\n",
    "- \"ReduceDimensions\"\n",
    "by/or select a subset of proteins to base analysis on\n",
    "- Lasso and Ridge\n",
    "- \"Subset selection\"\n",
    "perform Prinicpal Component Analysis (PCA)\n",
    "or perform Linear Discriminant Analysis (LDA)\n",
    "perform hierarchical clustering\n",
    "decision tree & x-fold cross validation\n",
    "ROC curves\n",
    "\"Ensemble Clustering\"\n",
    "kmeans clustering\n",
    "\n",
    "__Need to look at variance!!!__ (chromosome 5 genes varry to a much larger extent than others)\n",
    "\n",
    "- review Kernals below - do they all use scikitlearn\n",
    "    - Clustering proteins (Pete Bleackley)\n",
    "        - https://www.kaggle.com/petebleackley/clustering-proteins\n",
    "        - heirarchical clustering into 8 groups\n",
    "        - beautifully colored bar graphs (need better ordering)\n",
    "    - Data Visualization of Molecular Subtype (Jason Huang)\n",
    "        - https://www.kaggle.com/jwlzdh1/data-visualization-of-molecular-subtype\n",
    "    - Hierarchical Clustering (Anthony Leo, Detriot, Software engineer @ Plex Systems)\n",
    "        - https://www.kaggle.com/anthonyleo/hierarchical-clustering\n",
    "    - Breast Cancer Sub-Characterization w/ Genomic Data (Jason T. Smith, Antarctica)\n",
    "        - https://www.kaggle.com/jasontsmith2718/breast-cancer-sub-characterization-w-genomic-data\n",
    "\n",
    "       \n",
    "    \n",
    "(3) Show correlation between PAM50(35) cluster and T&N\n",
    "    Show correlation between All preotein clusters and T&N\n",
    "    (see one of the kernals with nice bar graphs)\n",
    "    which clusters predict low T&N, which clusters predict high T&N  \n",
    "    \n",
    "   pick out new kinases to target in non HER2 mutants\n",
    "\n",
    "XPredict\n",
    "- AJCC stage/score (includes tumor size (T) and spread to LNs (N))\n",
    "- ER, PR, Her2, basal...\n",
    "- delete the top 8 then determine which clusters determine the 8\n",
    "- identify small molecule drug targets: proteins that have the string 'ase' \n",
    "- identify antibody targets: proteins expressed on the surface of cells\n",
    "- identify CRISPR targets: proteins that are excessively expressed that could be targeted for knockdown\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "# import numpy.linalg\n",
    "\n",
    "import pandas as pd\n",
    "import matplotlib # Do I need to import this separately \n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import seaborn as sns\n",
    "\n",
    "# import re # could import with a comma ', re' after sklearn\n",
    "# import itertools\n",
    "# import sklearn\n",
    "\n",
    "# from sklearn import metrics\n",
    "# from sklearn import preprocessing\n",
    "\n",
    "# from sklearn.preprocessing import Imputer # Bewarem imputer.transform returns a numpy array, not a dataframe\n",
    "# from sklearn.preprocessing import StandardScaler\n",
    "# from sklearn.preprocessing import LabelEncoder\n",
    "\n",
    "# from sklearn.decomposition import PCA\n",
    "\n",
    "# from sklearn.cluster import KMeans\n",
    "# from sklearn.cluster import DBSCAN\n",
    "# from sklearn.cluster import SpectralClustering\n",
    "# from sklearn.cluster import MeanShift\n",
    "\n",
    "# from sklearn.metrics import adjusted_rand_score as rn\n",
    "# from sklearn.metrics import adjusted_mutual_info_score as mi\n",
    "\n",
    "# from scipy.cluster.hierarchy import dendrogram # do I need to import scipy?\n",
    "\n",
    "# from collections import defaultdict\n",
    "# from collections import OrderedDict\n",
    "\n",
    "# from mpl_toolkits.mplot3d import Axes3D\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import files as dataframes\n",
    "ProtExpressDF = pd.read_csv('origdata/77_cancer_proteomes_CPTAC_itraq.csv', header = 0, low_memory = False)\n",
    "ClinMetaDF = pd.read_csv('origdata/clinical_data_breast_cancer.csv', header = 0, low_memory = False)\n",
    "PAM50DF = pd.read_csv('origdata/PAM50_proteins.csv', header = 0, low_memory = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RefSeq_accession_number</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>AO-A12D.01TCGA</th>\n",
       "      <th>C8-A131.01TCGA</th>\n",
       "      <th>AO-A12B.01TCGA</th>\n",
       "      <th>BH-A18Q.02TCGA</th>\n",
       "      <th>C8-A130.02TCGA</th>\n",
       "      <th>C8-A138.03TCGA</th>\n",
       "      <th>E2-A154.03TCGA</th>\n",
       "      <th>...</th>\n",
       "      <th>AO-A12B.34TCGA</th>\n",
       "      <th>A2-A0SW.35TCGA</th>\n",
       "      <th>AO-A0JL.35TCGA</th>\n",
       "      <th>BH-A0BV.35TCGA</th>\n",
       "      <th>A2-A0YM.36TCGA</th>\n",
       "      <th>BH-A0C7.36TCGA</th>\n",
       "      <th>A2-A0SX.36TCGA</th>\n",
       "      <th>263d3f-I.CPTAC</th>\n",
       "      <th>blcdb9-I.CPTAC</th>\n",
       "      <th>c4155b-C.CPTAC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_958782</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1</td>\n",
       "      <td>1.096131</td>\n",
       "      <td>2.609943</td>\n",
       "      <td>-0.659828</td>\n",
       "      <td>0.195341</td>\n",
       "      <td>-0.494060</td>\n",
       "      <td>2.765081</td>\n",
       "      <td>0.862659</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.963904</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.106680</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.398560</td>\n",
       "      <td>0.598585</td>\n",
       "      <td>-0.191285</td>\n",
       "      <td>0.566975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_958785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1g</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.648742</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.938210</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.106680</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.658143</td>\n",
       "      <td>-0.547749</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.606697</td>\n",
       "      <td>-0.183918</td>\n",
       "      <td>0.578702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NP_958786</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1a</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.654285</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.500619</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.943919</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.106680</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NP_000436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1c</td>\n",
       "      <td>1.107561</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.632113</td>\n",
       "      <td>0.205377</td>\n",
       "      <td>-0.510459</td>\n",
       "      <td>2.797995</td>\n",
       "      <td>0.866423</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.106680</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NP_958781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1e</td>\n",
       "      <td>1.115180</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.640428</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.787023</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.503853</td>\n",
       "      <td>-0.106680</td>\n",
       "      <td>-0.062523</td>\n",
       "      <td>0.651264</td>\n",
       "      <td>-0.556675</td>\n",
       "      <td>-0.395581</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.167079</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NP_958780</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1f</td>\n",
       "      <td>1.107561</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.654285</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.938210</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.106680</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.658143</td>\n",
       "      <td>-0.547749</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.606697</td>\n",
       "      <td>-0.183918</td>\n",
       "      <td>0.578702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NP_958783</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1d</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.648742</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.500619</td>\n",
       "      <td>2.783366</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.943919</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.106680</td>\n",
       "      <td>-0.062523</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NP_958784</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1b</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.648742</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.500619</td>\n",
       "      <td>2.783366</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.943919</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.106680</td>\n",
       "      <td>-0.062523</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NP_112598</td>\n",
       "      <td>NaN</td>\n",
       "      <td>epiplakin</td>\n",
       "      <td>-1.517390</td>\n",
       "      <td>3.909313</td>\n",
       "      <td>-0.618256</td>\n",
       "      <td>-1.035760</td>\n",
       "      <td>-1.845366</td>\n",
       "      <td>2.205538</td>\n",
       "      <td>1.920171</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.252252</td>\n",
       "      <td>-1.626289</td>\n",
       "      <td>0.025189</td>\n",
       "      <td>-2.187600</td>\n",
       "      <td>-1.969534</td>\n",
       "      <td>0.679466</td>\n",
       "      <td>-2.504862</td>\n",
       "      <td>-0.602132</td>\n",
       "      <td>-0.340726</td>\n",
       "      <td>-0.205013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NP_001611</td>\n",
       "      <td>AHNAK</td>\n",
       "      <td>neuroblast differentiation-associated protein ...</td>\n",
       "      <td>0.482754</td>\n",
       "      <td>-1.045294</td>\n",
       "      <td>1.222003</td>\n",
       "      <td>-0.517226</td>\n",
       "      <td>-0.405503</td>\n",
       "      <td>0.749997</td>\n",
       "      <td>2.349197</td>\n",
       "      <td>...</td>\n",
       "      <td>1.325752</td>\n",
       "      <td>0.731148</td>\n",
       "      <td>-1.177327</td>\n",
       "      <td>0.709931</td>\n",
       "      <td>1.307036</td>\n",
       "      <td>0.487574</td>\n",
       "      <td>0.694810</td>\n",
       "      <td>2.778263</td>\n",
       "      <td>1.367330</td>\n",
       "      <td>3.215190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NP_076965</td>\n",
       "      <td>AHNAK</td>\n",
       "      <td>neuroblast differentiation-associated protein ...</td>\n",
       "      <td>0.261785</td>\n",
       "      <td>-0.037371</td>\n",
       "      <td>1.019685</td>\n",
       "      <td>-0.724639</td>\n",
       "      <td>-0.703971</td>\n",
       "      <td>-0.156974</td>\n",
       "      <td>1.581466</td>\n",
       "      <td>...</td>\n",
       "      <td>1.645505</td>\n",
       "      <td>0.358075</td>\n",
       "      <td>-3.466189</td>\n",
       "      <td>-0.503452</td>\n",
       "      <td>1.300158</td>\n",
       "      <td>0.721860</td>\n",
       "      <td>0.572663</td>\n",
       "      <td>3.846468</td>\n",
       "      <td>1.088442</td>\n",
       "      <td>3.480988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NP_001367</td>\n",
       "      <td>DYNC1H1</td>\n",
       "      <td>cytoplasmic dynein 1 heavy chain 1</td>\n",
       "      <td>0.924690</td>\n",
       "      <td>1.399627</td>\n",
       "      <td>-1.230752</td>\n",
       "      <td>-0.085671</td>\n",
       "      <td>-0.785968</td>\n",
       "      <td>1.507025</td>\n",
       "      <td>0.064821</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.574860</td>\n",
       "      <td>-1.246784</td>\n",
       "      <td>-0.241688</td>\n",
       "      <td>-0.254808</td>\n",
       "      <td>-0.325516</td>\n",
       "      <td>-2.553688</td>\n",
       "      <td>-1.211884</td>\n",
       "      <td>-1.024005</td>\n",
       "      <td>-0.328097</td>\n",
       "      <td>-0.547034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NP_036222</td>\n",
       "      <td>MACF1</td>\n",
       "      <td>microtubule-actin cross-linking factor 1</td>\n",
       "      <td>-0.580180</td>\n",
       "      <td>-0.134520</td>\n",
       "      <td>0.781338</td>\n",
       "      <td>-0.306467</td>\n",
       "      <td>-0.917163</td>\n",
       "      <td>-1.305315</td>\n",
       "      <td>0.395999</td>\n",
       "      <td>...</td>\n",
       "      <td>0.860398</td>\n",
       "      <td>-2.134441</td>\n",
       "      <td>-1.886906</td>\n",
       "      <td>-1.100197</td>\n",
       "      <td>-1.621011</td>\n",
       "      <td>-1.998094</td>\n",
       "      <td>-2.534654</td>\n",
       "      <td>0.374126</td>\n",
       "      <td>0.041297</td>\n",
       "      <td>-0.330095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NP_001138241</td>\n",
       "      <td>NaN</td>\n",
       "      <td>bullous pemphigoid antigen 1 isoform 2</td>\n",
       "      <td>-0.012520</td>\n",
       "      <td>0.756014</td>\n",
       "      <td>-1.092178</td>\n",
       "      <td>0.737293</td>\n",
       "      <td>-1.133634</td>\n",
       "      <td>-0.504402</td>\n",
       "      <td>-1.523327</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.226558</td>\n",
       "      <td>0.908036</td>\n",
       "      <td>-0.850796</td>\n",
       "      <td>0.080033</td>\n",
       "      <td>0.699415</td>\n",
       "      <td>-1.031941</td>\n",
       "      <td>-0.377705</td>\n",
       "      <td>1.458557</td>\n",
       "      <td>0.201263</td>\n",
       "      <td>0.879680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NP_001714</td>\n",
       "      <td>DST</td>\n",
       "      <td>bullous pemphigoid antigen 1 isoform 1e precursor</td>\n",
       "      <td>-0.187771</td>\n",
       "      <td>0.173119</td>\n",
       "      <td>-1.785046</td>\n",
       "      <td>2.041991</td>\n",
       "      <td>-0.520299</td>\n",
       "      <td>-0.182574</td>\n",
       "      <td>-2.343745</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.585506</td>\n",
       "      <td>-0.626067</td>\n",
       "      <td>-1.569793</td>\n",
       "      <td>0.036934</td>\n",
       "      <td>-0.451626</td>\n",
       "      <td>-0.871288</td>\n",
       "      <td>0.146636</td>\n",
       "      <td>4.617198</td>\n",
       "      <td>0.554872</td>\n",
       "      <td>2.054275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NP_002464</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>myosin-9</td>\n",
       "      <td>2.044771</td>\n",
       "      <td>1.593925</td>\n",
       "      <td>-1.266781</td>\n",
       "      <td>1.028341</td>\n",
       "      <td>0.109436</td>\n",
       "      <td>1.254682</td>\n",
       "      <td>0.418580</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.808964</td>\n",
       "      <td>-0.494205</td>\n",
       "      <td>0.317183</td>\n",
       "      <td>-0.821716</td>\n",
       "      <td>0.064279</td>\n",
       "      <td>-1.989169</td>\n",
       "      <td>-0.204911</td>\n",
       "      <td>-0.861746</td>\n",
       "      <td>-0.697492</td>\n",
       "      <td>-0.513809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NP_001073936</td>\n",
       "      <td>MYO5B</td>\n",
       "      <td>unconventional myosin-Vb</td>\n",
       "      <td>-0.065857</td>\n",
       "      <td>-0.875283</td>\n",
       "      <td>-2.281140</td>\n",
       "      <td>1.868032</td>\n",
       "      <td>-0.976200</td>\n",
       "      <td>0.464740</td>\n",
       "      <td>-2.776535</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.953792</td>\n",
       "      <td>-0.690390</td>\n",
       "      <td>-2.860222</td>\n",
       "      <td>-0.874760</td>\n",
       "      <td>-1.038612</td>\n",
       "      <td>-1.562991</td>\n",
       "      <td>-2.540612</td>\n",
       "      <td>-1.702789</td>\n",
       "      <td>-1.289997</td>\n",
       "      <td>-1.549642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NP_061198</td>\n",
       "      <td>MYO5C</td>\n",
       "      <td>unconventional myosin-Vc</td>\n",
       "      <td>-0.351592</td>\n",
       "      <td>-1.021006</td>\n",
       "      <td>0.016412</td>\n",
       "      <td>-0.881872</td>\n",
       "      <td>-1.517379</td>\n",
       "      <td>2.073881</td>\n",
       "      <td>1.442221</td>\n",
       "      <td>...</td>\n",
       "      <td>0.303686</td>\n",
       "      <td>-1.899662</td>\n",
       "      <td>-2.200879</td>\n",
       "      <td>1.233740</td>\n",
       "      <td>-0.777220</td>\n",
       "      <td>-1.246146</td>\n",
       "      <td>-1.214864</td>\n",
       "      <td>0.030678</td>\n",
       "      <td>-0.960594</td>\n",
       "      <td>-0.861692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NP_001074248</td>\n",
       "      <td>MYO1C</td>\n",
       "      <td>unconventional myosin-Ic isoform a</td>\n",
       "      <td>0.318932</td>\n",
       "      <td>-0.438111</td>\n",
       "      <td>0.088470</td>\n",
       "      <td>0.780782</td>\n",
       "      <td>-0.608855</td>\n",
       "      <td>-0.387373</td>\n",
       "      <td>2.443281</td>\n",
       "      <td>...</td>\n",
       "      <td>0.306541</td>\n",
       "      <td>1.708858</td>\n",
       "      <td>0.486729</td>\n",
       "      <td>1.548689</td>\n",
       "      <td>3.895733</td>\n",
       "      <td>-0.574525</td>\n",
       "      <td>2.857717</td>\n",
       "      <td>3.960050</td>\n",
       "      <td>1.791450</td>\n",
       "      <td>3.989133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NP_001074419</td>\n",
       "      <td>MYO1C</td>\n",
       "      <td>unconventional myosin-Ic isoform b</td>\n",
       "      <td>0.318932</td>\n",
       "      <td>-0.438111</td>\n",
       "      <td>0.088470</td>\n",
       "      <td>0.787473</td>\n",
       "      <td>-0.621974</td>\n",
       "      <td>-0.387373</td>\n",
       "      <td>2.435754</td>\n",
       "      <td>...</td>\n",
       "      <td>0.306541</td>\n",
       "      <td>1.737803</td>\n",
       "      <td>0.496148</td>\n",
       "      <td>1.568581</td>\n",
       "      <td>3.902612</td>\n",
       "      <td>-0.570062</td>\n",
       "      <td>2.887509</td>\n",
       "      <td>3.862694</td>\n",
       "      <td>1.791450</td>\n",
       "      <td>3.967634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NP_056009</td>\n",
       "      <td>MYO1D</td>\n",
       "      <td>unconventional myosin-Id</td>\n",
       "      <td>0.970408</td>\n",
       "      <td>1.642499</td>\n",
       "      <td>-0.906489</td>\n",
       "      <td>0.018035</td>\n",
       "      <td>0.847407</td>\n",
       "      <td>1.108397</td>\n",
       "      <td>1.634153</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.326480</td>\n",
       "      <td>-0.198319</td>\n",
       "      <td>0.631156</td>\n",
       "      <td>0.650256</td>\n",
       "      <td>0.857626</td>\n",
       "      <td>-0.840050</td>\n",
       "      <td>-0.207890</td>\n",
       "      <td>0.022565</td>\n",
       "      <td>-0.584885</td>\n",
       "      <td>-0.263645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NP_036355</td>\n",
       "      <td>MYO1B</td>\n",
       "      <td>unconventional myosin-Ib isoform 2</td>\n",
       "      <td>1.526638</td>\n",
       "      <td>2.925678</td>\n",
       "      <td>-2.383685</td>\n",
       "      <td>-0.758093</td>\n",
       "      <td>0.555498</td>\n",
       "      <td>-0.054574</td>\n",
       "      <td>-0.789467</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.421227</td>\n",
       "      <td>-0.809387</td>\n",
       "      <td>-0.150636</td>\n",
       "      <td>-1.875966</td>\n",
       "      <td>0.926413</td>\n",
       "      <td>-1.255071</td>\n",
       "      <td>-1.119529</td>\n",
       "      <td>-0.031521</td>\n",
       "      <td>-0.791156</td>\n",
       "      <td>-0.304688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NP_001155291</td>\n",
       "      <td>MYO1B</td>\n",
       "      <td>unconventional myosin-Ib isoform 1</td>\n",
       "      <td>1.526638</td>\n",
       "      <td>2.925678</td>\n",
       "      <td>-2.383685</td>\n",
       "      <td>-0.758093</td>\n",
       "      <td>0.555498</td>\n",
       "      <td>-0.083831</td>\n",
       "      <td>-0.759360</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.435502</td>\n",
       "      <td>-0.809387</td>\n",
       "      <td>-0.072143</td>\n",
       "      <td>-1.875966</td>\n",
       "      <td>0.926413</td>\n",
       "      <td>-1.255071</td>\n",
       "      <td>-1.119529</td>\n",
       "      <td>-0.001774</td>\n",
       "      <td>-0.786947</td>\n",
       "      <td>-0.291007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NP_002462</td>\n",
       "      <td>MYH6</td>\n",
       "      <td>myosin-6</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>-0.616218</td>\n",
       "      <td>-3.467331</td>\n",
       "      <td>-3.865953</td>\n",
       "      <td>-0.448141</td>\n",
       "      <td>0.801197</td>\n",
       "      <td>-0.883552</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.681800</td>\n",
       "      <td>-0.809387</td>\n",
       "      <td>0.552663</td>\n",
       "      <td>-1.594170</td>\n",
       "      <td>3.710007</td>\n",
       "      <td>2.551526</td>\n",
       "      <td>3.587624</td>\n",
       "      <td>-0.099129</td>\n",
       "      <td>-1.497321</td>\n",
       "      <td>-0.676024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NP_000248</td>\n",
       "      <td>MYH7</td>\n",
       "      <td>myosin-7</td>\n",
       "      <td>-0.507794</td>\n",
       "      <td>1.391531</td>\n",
       "      <td>-2.799406</td>\n",
       "      <td>-3.865953</td>\n",
       "      <td>-0.510459</td>\n",
       "      <td>0.512283</td>\n",
       "      <td>-1.496984</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.793142</td>\n",
       "      <td>-0.809387</td>\n",
       "      <td>0.552663</td>\n",
       "      <td>-1.594170</td>\n",
       "      <td>3.776502</td>\n",
       "      <td>2.687636</td>\n",
       "      <td>3.742543</td>\n",
       "      <td>-0.109946</td>\n",
       "      <td>-1.592038</td>\n",
       "      <td>-0.973093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NP_036467</td>\n",
       "      <td>MYO1F</td>\n",
       "      <td>unconventional myosin-If</td>\n",
       "      <td>0.082725</td>\n",
       "      <td>2.524937</td>\n",
       "      <td>-2.466829</td>\n",
       "      <td>1.924903</td>\n",
       "      <td>-0.464541</td>\n",
       "      <td>-0.080174</td>\n",
       "      <td>-0.665275</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.959502</td>\n",
       "      <td>-0.584257</td>\n",
       "      <td>0.310904</td>\n",
       "      <td>-0.251493</td>\n",
       "      <td>1.630337</td>\n",
       "      <td>-2.252463</td>\n",
       "      <td>1.394925</td>\n",
       "      <td>-0.794138</td>\n",
       "      <td>-1.562570</td>\n",
       "      <td>-0.922279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NP_004989</td>\n",
       "      <td>MYO1E</td>\n",
       "      <td>unconventional myosin-Ie</td>\n",
       "      <td>0.539901</td>\n",
       "      <td>2.184915</td>\n",
       "      <td>-2.641432</td>\n",
       "      <td>1.965048</td>\n",
       "      <td>-0.225110</td>\n",
       "      <td>0.739026</td>\n",
       "      <td>-0.085714</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.693220</td>\n",
       "      <td>-0.645364</td>\n",
       "      <td>0.678252</td>\n",
       "      <td>-1.206285</td>\n",
       "      <td>1.341430</td>\n",
       "      <td>-1.676787</td>\n",
       "      <td>-0.085743</td>\n",
       "      <td>0.065834</td>\n",
       "      <td>-0.659606</td>\n",
       "      <td>-0.109248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NP_001243024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>myosin-10 isoform 3</td>\n",
       "      <td>2.101918</td>\n",
       "      <td>0.513141</td>\n",
       "      <td>-0.673685</td>\n",
       "      <td>-0.821655</td>\n",
       "      <td>0.230791</td>\n",
       "      <td>1.327825</td>\n",
       "      <td>0.437397</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.009583</td>\n",
       "      <td>-1.970417</td>\n",
       "      <td>1.171189</td>\n",
       "      <td>-0.148720</td>\n",
       "      <td>0.942464</td>\n",
       "      <td>-1.089955</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.555315</td>\n",
       "      <td>-0.447019</td>\n",
       "      <td>0.334401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NP_149043</td>\n",
       "      <td>MYO1G</td>\n",
       "      <td>unconventional myosin-Ig</td>\n",
       "      <td>0.802777</td>\n",
       "      <td>1.294382</td>\n",
       "      <td>-1.943020</td>\n",
       "      <td>2.098863</td>\n",
       "      <td>-0.592456</td>\n",
       "      <td>-0.420288</td>\n",
       "      <td>-0.830864</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.314287</td>\n",
       "      <td>0.563908</td>\n",
       "      <td>0.832099</td>\n",
       "      <td>-0.009479</td>\n",
       "      <td>1.096089</td>\n",
       "      <td>-1.228296</td>\n",
       "      <td>2.440628</td>\n",
       "      <td>-0.731939</td>\n",
       "      <td>-0.597514</td>\n",
       "      <td>-0.193287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NP_067047</td>\n",
       "      <td>ZNF462</td>\n",
       "      <td>zinc finger protein 462</td>\n",
       "      <td>1.183757</td>\n",
       "      <td>-0.838852</td>\n",
       "      <td>-7.763116</td>\n",
       "      <td>1.101939</td>\n",
       "      <td>-0.149673</td>\n",
       "      <td>1.532625</td>\n",
       "      <td>-1.086775</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.362047</td>\n",
       "      <td>1.695993</td>\n",
       "      <td>1.723782</td>\n",
       "      <td>-0.248177</td>\n",
       "      <td>0.435730</td>\n",
       "      <td>-1.152432</td>\n",
       "      <td>1.823931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12523</th>\n",
       "      <td>NP_004953</td>\n",
       "      <td>GDF10</td>\n",
       "      <td>bone morphogenetic protein 3B precursor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.159808</td>\n",
       "      <td>1.742811</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.183128</td>\n",
       "      <td>-4.524277</td>\n",
       "      <td>-4.614364</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12524</th>\n",
       "      <td>NP_001034806</td>\n",
       "      <td>C4ORF29</td>\n",
       "      <td>uncharacterized protein C4orf29 precursor</td>\n",
       "      <td>1.275192</td>\n",
       "      <td>1.314621</td>\n",
       "      <td>0.529134</td>\n",
       "      <td>-3.414326</td>\n",
       "      <td>-1.619055</td>\n",
       "      <td>-0.826230</td>\n",
       "      <td>0.362129</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12525</th>\n",
       "      <td>NP_065119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>phospholipid-metabolizing enzyme A-C1</td>\n",
       "      <td>0.787538</td>\n",
       "      <td>0.662912</td>\n",
       "      <td>-1.541157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.404058</td>\n",
       "      <td>-1.331395</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.046218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.314631</td>\n",
       "      <td>-2.492898</td>\n",
       "      <td>-0.799152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12526</th>\n",
       "      <td>NP_110392</td>\n",
       "      <td>B3GNT4</td>\n",
       "      <td>UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosamin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.935944</td>\n",
       "      <td>3.350257</td>\n",
       "      <td>...</td>\n",
       "      <td>0.323671</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.236803</td>\n",
       "      <td>0.462260</td>\n",
       "      <td>1.978053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12527</th>\n",
       "      <td>NP_620138</td>\n",
       "      <td>NaN</td>\n",
       "      <td>zinc finger protein 653</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12528</th>\n",
       "      <td>NP_062558</td>\n",
       "      <td>IRGC</td>\n",
       "      <td>interferon-inducible GTPase 5</td>\n",
       "      <td>0.825636</td>\n",
       "      <td>0.136688</td>\n",
       "      <td>1.887156</td>\n",
       "      <td>2.895064</td>\n",
       "      <td>1.755930</td>\n",
       "      <td>-0.255716</td>\n",
       "      <td>2.277692</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.131002</td>\n",
       "      <td>-0.188313</td>\n",
       "      <td>1.363035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.369315</td>\n",
       "      <td>-0.854301</td>\n",
       "      <td>-0.081886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12529</th>\n",
       "      <td>NP_001005163</td>\n",
       "      <td>OR52D1</td>\n",
       "      <td>olfactory receptor 52D1</td>\n",
       "      <td>1.454252</td>\n",
       "      <td>1.002934</td>\n",
       "      <td>-4.900183</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-5.960853</td>\n",
       "      <td>-12.211343</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12530</th>\n",
       "      <td>NP_665813</td>\n",
       "      <td>CACNG6</td>\n",
       "      <td>voltage-dependent calcium channel gamma-6 subu...</td>\n",
       "      <td>0.840875</td>\n",
       "      <td>1.468441</td>\n",
       "      <td>-0.837202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.016968</td>\n",
       "      <td>-1.060431</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.571392</td>\n",
       "      <td>0.235550</td>\n",
       "      <td>-0.838292</td>\n",
       "      <td>0.016127</td>\n",
       "      <td>-2.399728</td>\n",
       "      <td>-1.214864</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12531</th>\n",
       "      <td>NP_005279</td>\n",
       "      <td>GPR12</td>\n",
       "      <td>G-protein coupled receptor 12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-13.547488</td>\n",
       "      <td>-0.061117</td>\n",
       "      <td>4.963021</td>\n",
       "      <td>-8.590966</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-5.310347</td>\n",
       "      <td>-1.882502</td>\n",
       "      <td>-3.156160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12532</th>\n",
       "      <td>NP_998761</td>\n",
       "      <td>FOXN4</td>\n",
       "      <td>forkhead box protein N4</td>\n",
       "      <td>0.139872</td>\n",
       "      <td>-0.446207</td>\n",
       "      <td>-2.228482</td>\n",
       "      <td>-4.896331</td>\n",
       "      <td>0.693253</td>\n",
       "      <td>2.563938</td>\n",
       "      <td>-2.878146</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12533</th>\n",
       "      <td>YP_003024032</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cytochrome c oxidase subunit III</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-4.708989</td>\n",
       "      <td>-1.648573</td>\n",
       "      <td>5.420164</td>\n",
       "      <td>-5.249079</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.773158</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12534</th>\n",
       "      <td>NP_001165887</td>\n",
       "      <td>NaN</td>\n",
       "      <td>inward rectifier potassium channel 13 isoform 2</td>\n",
       "      <td>0.932310</td>\n",
       "      <td>1.229616</td>\n",
       "      <td>0.756395</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.731144</td>\n",
       "      <td>0.979324</td>\n",
       "      <td>...</td>\n",
       "      <td>0.129535</td>\n",
       "      <td>-1.960769</td>\n",
       "      <td>-0.166335</td>\n",
       "      <td>-0.324428</td>\n",
       "      <td>0.087208</td>\n",
       "      <td>-5.240173</td>\n",
       "      <td>-0.842462</td>\n",
       "      <td>0.636445</td>\n",
       "      <td>-0.757479</td>\n",
       "      <td>0.574793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12535</th>\n",
       "      <td>NP_001010893</td>\n",
       "      <td>SLC10A5</td>\n",
       "      <td>sodium/bile acid cotransporter 5 precursor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.157540</td>\n",
       "      <td>0.693307</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12536</th>\n",
       "      <td>NP_872303</td>\n",
       "      <td>KRT40</td>\n",
       "      <td>keratin, type I cytoskeletal 40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.865684</td>\n",
       "      <td>-3.586976</td>\n",
       "      <td>5.489649</td>\n",
       "      <td>-5.109834</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12537</th>\n",
       "      <td>NP_000871</td>\n",
       "      <td>IL7</td>\n",
       "      <td>interleukin-7 isoform 1 precursor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.723911</td>\n",
       "      <td>3.120356</td>\n",
       "      <td>1.064511</td>\n",
       "      <td>2.044362</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.330643</td>\n",
       "      <td>4.185293</td>\n",
       "      <td>0.659414</td>\n",
       "      <td>1.843747</td>\n",
       "      <td>3.152830</td>\n",
       "      <td>-4.133448</td>\n",
       "      <td>1.582615</td>\n",
       "      <td>2.204948</td>\n",
       "      <td>-1.322622</td>\n",
       "      <td>1.329192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12538</th>\n",
       "      <td>NP_690872</td>\n",
       "      <td>SLC47A2</td>\n",
       "      <td>multidrug and toxin extrusion protein 2 isoform 1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.661080</td>\n",
       "      <td>2.488442</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.145072</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.396955</td>\n",
       "      <td>-2.042468</td>\n",
       "      <td>1.260788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12539</th>\n",
       "      <td>NP_683762</td>\n",
       "      <td>HUS1B</td>\n",
       "      <td>checkpoint protein HUS1B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.617194</td>\n",
       "      <td>-7.070558</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.126635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.600820</td>\n",
       "      <td>-6.451769</td>\n",
       "      <td>-1.578327</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12540</th>\n",
       "      <td>NP_060064</td>\n",
       "      <td>DIRAS2</td>\n",
       "      <td>GTP-binding protein Di-Ras2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.118070</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12541</th>\n",
       "      <td>NP_699174</td>\n",
       "      <td>ENPP6</td>\n",
       "      <td>ectonucleotide pyrophosphatase/phosphodiestera...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.583405</td>\n",
       "      <td>0.332467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.689267</td>\n",
       "      <td>-1.459435</td>\n",
       "      <td>-1.829121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12542</th>\n",
       "      <td>NP_001164045</td>\n",
       "      <td>FAM46D</td>\n",
       "      <td>protein FAM46D</td>\n",
       "      <td>-1.231655</td>\n",
       "      <td>-4.279550</td>\n",
       "      <td>-4.462290</td>\n",
       "      <td>-11.864761</td>\n",
       "      <td>0.115996</td>\n",
       "      <td>6.755019</td>\n",
       "      <td>-7.168406</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12543</th>\n",
       "      <td>NP_001193600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein RIC-3 isoform c precursor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.048144</td>\n",
       "      <td>1.457462</td>\n",
       "      <td>-3.250913</td>\n",
       "      <td>-3.386203</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.596070</td>\n",
       "      <td>-4.539976</td>\n",
       "      <td>-1.086936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.162522</td>\n",
       "      <td>-4.357763</td>\n",
       "      <td>-1.543779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12544</th>\n",
       "      <td>NP_061134</td>\n",
       "      <td>BATF3</td>\n",
       "      <td>basic leucine zipper transcriptional factor AT...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.711825</td>\n",
       "      <td>-2.328692</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.749784</td>\n",
       "      <td>0.969143</td>\n",
       "      <td>0.860356</td>\n",
       "      <td>2.128858</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.520756</td>\n",
       "      <td>-2.386605</td>\n",
       "      <td>0.518115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12545</th>\n",
       "      <td>NP_932347</td>\n",
       "      <td>SCML4</td>\n",
       "      <td>sex comb on midleg-like protein 4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.881872</td>\n",
       "      <td>1.710012</td>\n",
       "      <td>-0.248402</td>\n",
       "      <td>-2.806642</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.903177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12546</th>\n",
       "      <td>NP_003593</td>\n",
       "      <td>FKBP6</td>\n",
       "      <td>peptidyl-prolyl cis-trans isomerase FKBP6 isof...</td>\n",
       "      <td>-0.340163</td>\n",
       "      <td>3.451902</td>\n",
       "      <td>-1.718531</td>\n",
       "      <td>2.527072</td>\n",
       "      <td>0.296389</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.447161</td>\n",
       "      <td>-1.053815</td>\n",
       "      <td>-1.202445</td>\n",
       "      <td>-1.219546</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12547</th>\n",
       "      <td>NP_997203</td>\n",
       "      <td>OTUD6A</td>\n",
       "      <td>OTU domain-containing protein 6A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-8.111243</td>\n",
       "      <td>-1.753529</td>\n",
       "      <td>4.707022</td>\n",
       "      <td>-4.733495</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12548</th>\n",
       "      <td>NP_001191293</td>\n",
       "      <td>NaN</td>\n",
       "      <td>protein FAM24B precursor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-16.029761</td>\n",
       "      <td>1.729692</td>\n",
       "      <td>4.107251</td>\n",
       "      <td>-9.584499</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.662350</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-8.020071</td>\n",
       "      <td>-3.093822</td>\n",
       "      <td>-4.602418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12549</th>\n",
       "      <td>NP_775791</td>\n",
       "      <td>NaN</td>\n",
       "      <td>putative uncharacterized protein C9orf62</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.046065</td>\n",
       "      <td>-0.425182</td>\n",
       "      <td>-3.203370</td>\n",
       "      <td>-4.786183</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.049608</td>\n",
       "      <td>-0.646977</td>\n",
       "      <td>0.240590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12550</th>\n",
       "      <td>NP_004065</td>\n",
       "      <td>COX8A</td>\n",
       "      <td>cytochrome c oxidase subunit 8A, mitochondrial</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.778435</td>\n",
       "      <td>-0.149673</td>\n",
       "      <td>1.971481</td>\n",
       "      <td>-3.103949</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12551</th>\n",
       "      <td>NP_068752</td>\n",
       "      <td>MIIP</td>\n",
       "      <td>migration and invasion-inhibitory protein</td>\n",
       "      <td>-0.633517</td>\n",
       "      <td>4.840325</td>\n",
       "      <td>-1.965192</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.019861</td>\n",
       "      <td>-1.718327</td>\n",
       "      <td>-0.369183</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12552</th>\n",
       "      <td>NP_219494</td>\n",
       "      <td>KIAA1737</td>\n",
       "      <td>uncharacterized protein KIAA1737</td>\n",
       "      <td>12.666488</td>\n",
       "      <td>0.140736</td>\n",
       "      <td>-2.854835</td>\n",
       "      <td>-3.069752</td>\n",
       "      <td>-0.047997</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-6.002860</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12553 rows  86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      RefSeq_accession_number gene_symbol  \\\n",
       "0                   NP_958782        PLEC   \n",
       "1                   NP_958785         NaN   \n",
       "2                   NP_958786        PLEC   \n",
       "3                   NP_000436         NaN   \n",
       "4                   NP_958781         NaN   \n",
       "5                   NP_958780        PLEC   \n",
       "6                   NP_958783        PLEC   \n",
       "7                   NP_958784         NaN   \n",
       "8                   NP_112598         NaN   \n",
       "9                   NP_001611       AHNAK   \n",
       "10                  NP_076965       AHNAK   \n",
       "11                  NP_001367     DYNC1H1   \n",
       "12                  NP_036222       MACF1   \n",
       "13               NP_001138241         NaN   \n",
       "14                  NP_001714         DST   \n",
       "15                  NP_002464        MYH9   \n",
       "16               NP_001073936       MYO5B   \n",
       "17                  NP_061198       MYO5C   \n",
       "18               NP_001074248       MYO1C   \n",
       "19               NP_001074419       MYO1C   \n",
       "20                  NP_056009       MYO1D   \n",
       "21                  NP_036355       MYO1B   \n",
       "22               NP_001155291       MYO1B   \n",
       "23                  NP_002462        MYH6   \n",
       "24                  NP_000248        MYH7   \n",
       "25                  NP_036467       MYO1F   \n",
       "26                  NP_004989       MYO1E   \n",
       "27               NP_001243024         NaN   \n",
       "28                  NP_149043       MYO1G   \n",
       "29                  NP_067047      ZNF462   \n",
       "...                       ...         ...   \n",
       "12523               NP_004953       GDF10   \n",
       "12524            NP_001034806     C4ORF29   \n",
       "12525               NP_065119         NaN   \n",
       "12526               NP_110392      B3GNT4   \n",
       "12527               NP_620138         NaN   \n",
       "12528               NP_062558        IRGC   \n",
       "12529            NP_001005163      OR52D1   \n",
       "12530               NP_665813      CACNG6   \n",
       "12531               NP_005279       GPR12   \n",
       "12532               NP_998761       FOXN4   \n",
       "12533            YP_003024032         NaN   \n",
       "12534            NP_001165887         NaN   \n",
       "12535            NP_001010893     SLC10A5   \n",
       "12536               NP_872303       KRT40   \n",
       "12537               NP_000871         IL7   \n",
       "12538               NP_690872     SLC47A2   \n",
       "12539               NP_683762       HUS1B   \n",
       "12540               NP_060064      DIRAS2   \n",
       "12541               NP_699174       ENPP6   \n",
       "12542            NP_001164045      FAM46D   \n",
       "12543            NP_001193600         NaN   \n",
       "12544               NP_061134       BATF3   \n",
       "12545               NP_932347       SCML4   \n",
       "12546               NP_003593       FKBP6   \n",
       "12547               NP_997203      OTUD6A   \n",
       "12548            NP_001191293         NaN   \n",
       "12549               NP_775791         NaN   \n",
       "12550               NP_004065       COX8A   \n",
       "12551               NP_068752        MIIP   \n",
       "12552               NP_219494    KIAA1737   \n",
       "\n",
       "                                               gene_name  AO-A12D.01TCGA  \\\n",
       "0                                      plectin isoform 1        1.096131   \n",
       "1                                     plectin isoform 1g        1.111370   \n",
       "2                                     plectin isoform 1a        1.111370   \n",
       "3                                     plectin isoform 1c        1.107561   \n",
       "4                                     plectin isoform 1e        1.115180   \n",
       "5                                     plectin isoform 1f        1.107561   \n",
       "6                                     plectin isoform 1d        1.111370   \n",
       "7                                     plectin isoform 1b        1.111370   \n",
       "8                                              epiplakin       -1.517390   \n",
       "9      neuroblast differentiation-associated protein ...        0.482754   \n",
       "10     neuroblast differentiation-associated protein ...        0.261785   \n",
       "11                    cytoplasmic dynein 1 heavy chain 1        0.924690   \n",
       "12              microtubule-actin cross-linking factor 1       -0.580180   \n",
       "13                bullous pemphigoid antigen 1 isoform 2       -0.012520   \n",
       "14     bullous pemphigoid antigen 1 isoform 1e precursor       -0.187771   \n",
       "15                                              myosin-9        2.044771   \n",
       "16                              unconventional myosin-Vb       -0.065857   \n",
       "17                              unconventional myosin-Vc       -0.351592   \n",
       "18                    unconventional myosin-Ic isoform a        0.318932   \n",
       "19                    unconventional myosin-Ic isoform b        0.318932   \n",
       "20                              unconventional myosin-Id        0.970408   \n",
       "21                    unconventional myosin-Ib isoform 2        1.526638   \n",
       "22                    unconventional myosin-Ib isoform 1        1.526638   \n",
       "23                                              myosin-6        1.111370   \n",
       "24                                              myosin-7       -0.507794   \n",
       "25                              unconventional myosin-If        0.082725   \n",
       "26                              unconventional myosin-Ie        0.539901   \n",
       "27                                   myosin-10 isoform 3        2.101918   \n",
       "28                              unconventional myosin-Ig        0.802777   \n",
       "29                               zinc finger protein 462        1.183757   \n",
       "...                                                  ...             ...   \n",
       "12523            bone morphogenetic protein 3B precursor             NaN   \n",
       "12524          uncharacterized protein C4orf29 precursor        1.275192   \n",
       "12525              phospholipid-metabolizing enzyme A-C1        0.787538   \n",
       "12526  UDP-GlcNAc:betaGal beta-1,3-N-acetylglucosamin...             NaN   \n",
       "12527                            zinc finger protein 653             NaN   \n",
       "12528                      interferon-inducible GTPase 5        0.825636   \n",
       "12529                            olfactory receptor 52D1        1.454252   \n",
       "12530  voltage-dependent calcium channel gamma-6 subu...        0.840875   \n",
       "12531                      G-protein coupled receptor 12             NaN   \n",
       "12532                            forkhead box protein N4        0.139872   \n",
       "12533                   cytochrome c oxidase subunit III             NaN   \n",
       "12534    inward rectifier potassium channel 13 isoform 2        0.932310   \n",
       "12535         sodium/bile acid cotransporter 5 precursor             NaN   \n",
       "12536                    keratin, type I cytoskeletal 40             NaN   \n",
       "12537                  interleukin-7 isoform 1 precursor             NaN   \n",
       "12538  multidrug and toxin extrusion protein 2 isoform 1             NaN   \n",
       "12539                           checkpoint protein HUS1B             NaN   \n",
       "12540                        GTP-binding protein Di-Ras2             NaN   \n",
       "12541  ectonucleotide pyrophosphatase/phosphodiestera...             NaN   \n",
       "12542                                     protein FAM46D       -1.231655   \n",
       "12543                  protein RIC-3 isoform c precursor             NaN   \n",
       "12544  basic leucine zipper transcriptional factor AT...             NaN   \n",
       "12545                  sex comb on midleg-like protein 4             NaN   \n",
       "12546  peptidyl-prolyl cis-trans isomerase FKBP6 isof...       -0.340163   \n",
       "12547                   OTU domain-containing protein 6A             NaN   \n",
       "12548                           protein FAM24B precursor             NaN   \n",
       "12549           putative uncharacterized protein C9orf62             NaN   \n",
       "12550     cytochrome c oxidase subunit 8A, mitochondrial             NaN   \n",
       "12551          migration and invasion-inhibitory protein       -0.633517   \n",
       "12552                   uncharacterized protein KIAA1737       12.666488   \n",
       "\n",
       "       C8-A131.01TCGA  AO-A12B.01TCGA  BH-A18Q.02TCGA  C8-A130.02TCGA  \\\n",
       "0            2.609943       -0.659828        0.195341       -0.494060   \n",
       "1            2.650422       -0.648742        0.215413       -0.503899   \n",
       "2            2.650422       -0.654285        0.215413       -0.500619   \n",
       "3            2.646374       -0.632113        0.205377       -0.510459   \n",
       "4            2.646374       -0.640428        0.215413       -0.503899   \n",
       "5            2.646374       -0.654285        0.215413       -0.503899   \n",
       "6            2.650422       -0.648742        0.215413       -0.500619   \n",
       "7            2.650422       -0.648742        0.215413       -0.500619   \n",
       "8            3.909313       -0.618256       -1.035760       -1.845366   \n",
       "9           -1.045294        1.222003       -0.517226       -0.405503   \n",
       "10          -0.037371        1.019685       -0.724639       -0.703971   \n",
       "11           1.399627       -1.230752       -0.085671       -0.785968   \n",
       "12          -0.134520        0.781338       -0.306467       -0.917163   \n",
       "13           0.756014       -1.092178        0.737293       -1.133634   \n",
       "14           0.173119       -1.785046        2.041991       -0.520299   \n",
       "15           1.593925       -1.266781        1.028341        0.109436   \n",
       "16          -0.875283       -2.281140        1.868032       -0.976200   \n",
       "17          -1.021006        0.016412       -0.881872       -1.517379   \n",
       "18          -0.438111        0.088470        0.780782       -0.608855   \n",
       "19          -0.438111        0.088470        0.787473       -0.621974   \n",
       "20           1.642499       -0.906489        0.018035        0.847407   \n",
       "21           2.925678       -2.383685       -0.758093        0.555498   \n",
       "22           2.925678       -2.383685       -0.758093        0.555498   \n",
       "23          -0.616218       -3.467331       -3.865953       -0.448141   \n",
       "24           1.391531       -2.799406       -3.865953       -0.510459   \n",
       "25           2.524937       -2.466829        1.924903       -0.464541   \n",
       "26           2.184915       -2.641432        1.965048       -0.225110   \n",
       "27           0.513141       -0.673685       -0.821655        0.230791   \n",
       "28           1.294382       -1.943020        2.098863       -0.592456   \n",
       "29          -0.838852       -7.763116        1.101939       -0.149673   \n",
       "...               ...             ...             ...             ...   \n",
       "12523             NaN             NaN       -2.159808        1.742811   \n",
       "12524        1.314621        0.529134       -3.414326       -1.619055   \n",
       "12525        0.662912       -1.541157             NaN             NaN   \n",
       "12526             NaN             NaN             NaN             NaN   \n",
       "12527             NaN             NaN             NaN             NaN   \n",
       "12528        0.136688        1.887156        2.895064        1.755930   \n",
       "12529        1.002934       -4.900183             NaN             NaN   \n",
       "12530        1.468441       -0.837202             NaN             NaN   \n",
       "12531             NaN             NaN      -13.547488       -0.061117   \n",
       "12532       -0.446207       -2.228482       -4.896331        0.693253   \n",
       "12533             NaN             NaN       -4.708989       -1.648573   \n",
       "12534        1.229616        0.756395             NaN             NaN   \n",
       "12535             NaN             NaN             NaN             NaN   \n",
       "12536             NaN             NaN       -2.865684       -3.586976   \n",
       "12537             NaN             NaN        0.723911        3.120356   \n",
       "12538             NaN             NaN             NaN             NaN   \n",
       "12539             NaN             NaN             NaN             NaN   \n",
       "12540             NaN             NaN             NaN             NaN   \n",
       "12541             NaN             NaN        0.583405        0.332467   \n",
       "12542       -4.279550       -4.462290      -11.864761        0.115996   \n",
       "12543             NaN             NaN        0.048144        1.457462   \n",
       "12544             NaN             NaN             NaN             NaN   \n",
       "12545             NaN             NaN       -0.881872        1.710012   \n",
       "12546        3.451902       -1.718531        2.527072        0.296389   \n",
       "12547             NaN             NaN       -8.111243       -1.753529   \n",
       "12548             NaN             NaN      -16.029761        1.729692   \n",
       "12549             NaN             NaN       -2.046065       -0.425182   \n",
       "12550             NaN             NaN       -1.778435       -0.149673   \n",
       "12551        4.840325       -1.965192             NaN             NaN   \n",
       "12552        0.140736       -2.854835       -3.069752       -0.047997   \n",
       "\n",
       "       C8-A138.03TCGA  E2-A154.03TCGA       ...        AO-A12B.34TCGA  \\\n",
       "0            2.765081        0.862659       ...             -0.963904   \n",
       "1            2.779709        0.870186       ...             -0.938210   \n",
       "2            2.779709        0.870186       ...             -0.943919   \n",
       "3            2.797995        0.866423       ...             -0.935355   \n",
       "4            2.787023        0.870186       ...             -0.935355   \n",
       "5            2.779709        0.870186       ...             -0.938210   \n",
       "6            2.783366        0.870186       ...             -0.943919   \n",
       "7            2.783366        0.870186       ...             -0.943919   \n",
       "8            2.205538        1.920171       ...             -1.252252   \n",
       "9            0.749997        2.349197       ...              1.325752   \n",
       "10          -0.156974        1.581466       ...              1.645505   \n",
       "11           1.507025        0.064821       ...             -1.574860   \n",
       "12          -1.305315        0.395999       ...              0.860398   \n",
       "13          -0.504402       -1.523327       ...             -1.226558   \n",
       "14          -0.182574       -2.343745       ...             -2.585506   \n",
       "15           1.254682        0.418580       ...             -1.808964   \n",
       "16           0.464740       -2.776535       ...             -2.953792   \n",
       "17           2.073881        1.442221       ...              0.303686   \n",
       "18          -0.387373        2.443281       ...              0.306541   \n",
       "19          -0.387373        2.435754       ...              0.306541   \n",
       "20           1.108397        1.634153       ...             -1.326480   \n",
       "21          -0.054574       -0.789467       ...             -4.421227   \n",
       "22          -0.083831       -0.759360       ...             -4.435502   \n",
       "23           0.801197       -0.883552       ...             -3.681800   \n",
       "24           0.512283       -1.496984       ...             -3.793142   \n",
       "25          -0.080174       -0.665275       ...             -2.959502   \n",
       "26           0.739026       -0.085714       ...             -3.693220   \n",
       "27           1.327825        0.437397       ...             -1.009583   \n",
       "28          -0.420288       -0.830864       ...             -2.314287   \n",
       "29           1.532625       -1.086775       ...             -3.362047   \n",
       "...               ...             ...       ...                   ...   \n",
       "12523             NaN             NaN       ...                   NaN   \n",
       "12524       -0.826230        0.362129       ...                   NaN   \n",
       "12525       -1.404058       -1.331395       ...             -9.046218   \n",
       "12526       -0.935944        3.350257       ...              0.323671   \n",
       "12527             NaN             NaN       ...                   NaN   \n",
       "12528       -0.255716        2.277692       ...                   NaN   \n",
       "12529       -5.960853      -12.211343       ...                   NaN   \n",
       "12530        1.016968       -1.060431       ...                   NaN   \n",
       "12531        4.963021       -8.590966       ...                   NaN   \n",
       "12532        2.563938       -2.878146       ...                   NaN   \n",
       "12533        5.420164       -5.249079       ...             -3.773158   \n",
       "12534       -0.731144        0.979324       ...              0.129535   \n",
       "12535        0.157540        0.693307       ...                   NaN   \n",
       "12536        5.489649       -5.109834       ...                   NaN   \n",
       "12537        1.064511        2.044362       ...             -3.330643   \n",
       "12538        3.661080        2.488442       ...             -3.145072   \n",
       "12539        3.617194       -7.070558       ...             -1.126635   \n",
       "12540             NaN             NaN       ...             -1.118070   \n",
       "12541             NaN             NaN       ...                   NaN   \n",
       "12542        6.755019       -7.168406       ...                   NaN   \n",
       "12543       -3.250913       -3.386203       ...                   NaN   \n",
       "12544        1.711825       -2.328692       ...             -0.749784   \n",
       "12545       -0.248402       -2.806642       ...             -1.903177   \n",
       "12546             NaN             NaN       ...             -0.447161   \n",
       "12547        4.707022       -4.733495       ...                   NaN   \n",
       "12548        4.107251       -9.584499       ...             -6.662350   \n",
       "12549       -3.203370       -4.786183       ...                   NaN   \n",
       "12550        1.971481       -3.103949       ...                   NaN   \n",
       "12551             NaN             NaN       ...                   NaN   \n",
       "12552             NaN             NaN       ...             -6.002860   \n",
       "\n",
       "       A2-A0SW.35TCGA  AO-A0JL.35TCGA  BH-A0BV.35TCGA  A2-A0YM.36TCGA  \\\n",
       "0           -0.487772       -0.106680       -0.065838        0.655850   \n",
       "1           -0.487772       -0.106680       -0.055893        0.658143   \n",
       "2           -0.487772       -0.106680       -0.065838        0.655850   \n",
       "3           -0.487772       -0.106680       -0.055893        0.655850   \n",
       "4           -0.503853       -0.106680       -0.062523        0.651264   \n",
       "5           -0.487772       -0.106680       -0.055893        0.658143   \n",
       "6           -0.487772       -0.106680       -0.062523        0.655850   \n",
       "7           -0.487772       -0.106680       -0.062523        0.655850   \n",
       "8           -1.626289        0.025189       -2.187600       -1.969534   \n",
       "9            0.731148       -1.177327        0.709931        1.307036   \n",
       "10           0.358075       -3.466189       -0.503452        1.300158   \n",
       "11          -1.246784       -0.241688       -0.254808       -0.325516   \n",
       "12          -2.134441       -1.886906       -1.100197       -1.621011   \n",
       "13           0.908036       -0.850796        0.080033        0.699415   \n",
       "14          -0.626067       -1.569793        0.036934       -0.451626   \n",
       "15          -0.494205        0.317183       -0.821716        0.064279   \n",
       "16          -0.690390       -2.860222       -0.874760       -1.038612   \n",
       "17          -1.899662       -2.200879        1.233740       -0.777220   \n",
       "18           1.708858        0.486729        1.548689        3.895733   \n",
       "19           1.737803        0.496148        1.568581        3.902612   \n",
       "20          -0.198319        0.631156        0.650256        0.857626   \n",
       "21          -0.809387       -0.150636       -1.875966        0.926413   \n",
       "22          -0.809387       -0.072143       -1.875966        0.926413   \n",
       "23          -0.809387        0.552663       -1.594170        3.710007   \n",
       "24          -0.809387        0.552663       -1.594170        3.776502   \n",
       "25          -0.584257        0.310904       -0.251493        1.630337   \n",
       "26          -0.645364        0.678252       -1.206285        1.341430   \n",
       "27          -1.970417        1.171189       -0.148720        0.942464   \n",
       "28           0.563908        0.832099       -0.009479        1.096089   \n",
       "29           1.695993        1.723782       -0.248177        0.435730   \n",
       "...               ...             ...             ...             ...   \n",
       "12523       -4.183128       -4.524277       -4.614364             NaN   \n",
       "12524             NaN             NaN             NaN             NaN   \n",
       "12525             NaN             NaN             NaN             NaN   \n",
       "12526             NaN             NaN             NaN             NaN   \n",
       "12527             NaN             NaN             NaN             NaN   \n",
       "12528       -1.131002       -0.188313        1.363035             NaN   \n",
       "12529             NaN             NaN             NaN             NaN   \n",
       "12530       -0.571392        0.235550       -0.838292        0.016127   \n",
       "12531             NaN             NaN             NaN             NaN   \n",
       "12532             NaN             NaN             NaN             NaN   \n",
       "12533             NaN             NaN             NaN             NaN   \n",
       "12534       -1.960769       -0.166335       -0.324428        0.087208   \n",
       "12535             NaN             NaN             NaN             NaN   \n",
       "12536             NaN             NaN             NaN             NaN   \n",
       "12537        4.185293        0.659414        1.843747        3.152830   \n",
       "12538             NaN             NaN             NaN             NaN   \n",
       "12539             NaN             NaN             NaN        0.600820   \n",
       "12540             NaN             NaN             NaN             NaN   \n",
       "12541             NaN             NaN             NaN             NaN   \n",
       "12542             NaN             NaN             NaN             NaN   \n",
       "12543        0.596070       -4.539976       -1.086936             NaN   \n",
       "12544        0.969143        0.860356        2.128858             NaN   \n",
       "12545             NaN             NaN             NaN             NaN   \n",
       "12546       -1.053815       -1.202445       -1.219546             NaN   \n",
       "12547             NaN             NaN             NaN             NaN   \n",
       "12548             NaN             NaN             NaN             NaN   \n",
       "12549             NaN             NaN             NaN             NaN   \n",
       "12550             NaN             NaN             NaN             NaN   \n",
       "12551             NaN             NaN             NaN             NaN   \n",
       "12552             NaN             NaN             NaN             NaN   \n",
       "\n",
       "       BH-A0C7.36TCGA  A2-A0SX.36TCGA  263d3f-I.CPTAC  blcdb9-I.CPTAC  \\\n",
       "0           -0.552212       -0.398560        0.598585       -0.191285   \n",
       "1           -0.547749       -0.392601        0.606697       -0.183918   \n",
       "2           -0.552212       -0.392601        0.603993       -0.186022   \n",
       "3           -0.552212       -0.392601        0.603993       -0.186022   \n",
       "4           -0.556675       -0.395581        0.603993       -0.167079   \n",
       "5           -0.547749       -0.392601        0.606697       -0.183918   \n",
       "6           -0.552212       -0.392601        0.603993       -0.186022   \n",
       "7           -0.552212       -0.392601        0.603993       -0.186022   \n",
       "8            0.679466       -2.504862       -0.602132       -0.340726   \n",
       "9            0.487574        0.694810        2.778263        1.367330   \n",
       "10           0.721860        0.572663        3.846468        1.088442   \n",
       "11          -2.553688       -1.211884       -1.024005       -0.328097   \n",
       "12          -1.998094       -2.534654        0.374126        0.041297   \n",
       "13          -1.031941       -0.377705        1.458557        0.201263   \n",
       "14          -0.871288        0.146636        4.617198        0.554872   \n",
       "15          -1.989169       -0.204911       -0.861746       -0.697492   \n",
       "16          -1.562991       -2.540612       -1.702789       -1.289997   \n",
       "17          -1.246146       -1.214864        0.030678       -0.960594   \n",
       "18          -0.574525        2.857717        3.960050        1.791450   \n",
       "19          -0.570062        2.887509        3.862694        1.791450   \n",
       "20          -0.840050       -0.207890        0.022565       -0.584885   \n",
       "21          -1.255071       -1.119529       -0.031521       -0.791156   \n",
       "22          -1.255071       -1.119529       -0.001774       -0.786947   \n",
       "23           2.551526        3.587624       -0.099129       -1.497321   \n",
       "24           2.687636        3.742543       -0.109946       -1.592038   \n",
       "25          -2.252463        1.394925       -0.794138       -1.562570   \n",
       "26          -1.676787       -0.085743        0.065834       -0.659606   \n",
       "27          -1.089955       -0.392601        0.555315       -0.447019   \n",
       "28          -1.228296        2.440628       -0.731939       -0.597514   \n",
       "29          -1.152432        1.823931             NaN             NaN   \n",
       "...               ...             ...             ...             ...   \n",
       "12523             NaN             NaN             NaN             NaN   \n",
       "12524             NaN             NaN             NaN             NaN   \n",
       "12525             NaN             NaN        0.314631       -2.492898   \n",
       "12526             NaN             NaN        1.236803        0.462260   \n",
       "12527             NaN             NaN             NaN             NaN   \n",
       "12528             NaN             NaN        1.369315       -0.854301   \n",
       "12529             NaN             NaN             NaN             NaN   \n",
       "12530       -2.399728       -1.214864             NaN             NaN   \n",
       "12531             NaN             NaN       -5.310347       -1.882502   \n",
       "12532             NaN             NaN             NaN             NaN   \n",
       "12533             NaN             NaN             NaN             NaN   \n",
       "12534       -5.240173       -0.842462        0.636445       -0.757479   \n",
       "12535             NaN             NaN             NaN             NaN   \n",
       "12536             NaN             NaN             NaN             NaN   \n",
       "12537       -4.133448        1.582615        2.204948       -1.322622   \n",
       "12538             NaN             NaN        2.396955       -2.042468   \n",
       "12539       -6.451769       -1.578327             NaN             NaN   \n",
       "12540             NaN             NaN             NaN             NaN   \n",
       "12541             NaN             NaN       -1.689267       -1.459435   \n",
       "12542             NaN             NaN             NaN             NaN   \n",
       "12543             NaN             NaN       -2.162522       -4.357763   \n",
       "12544             NaN             NaN        1.520756       -2.386605   \n",
       "12545             NaN             NaN             NaN             NaN   \n",
       "12546             NaN             NaN             NaN             NaN   \n",
       "12547             NaN             NaN             NaN             NaN   \n",
       "12548             NaN             NaN       -8.020071       -3.093822   \n",
       "12549             NaN             NaN        0.049608       -0.646977   \n",
       "12550             NaN             NaN             NaN             NaN   \n",
       "12551             NaN             NaN        0.019861       -1.718327   \n",
       "12552             NaN             NaN             NaN             NaN   \n",
       "\n",
       "       c4155b-C.CPTAC  \n",
       "0            0.566975  \n",
       "1            0.578702  \n",
       "2            0.576747  \n",
       "3            0.576747  \n",
       "4            0.576747  \n",
       "5            0.578702  \n",
       "6            0.576747  \n",
       "7            0.576747  \n",
       "8           -0.205013  \n",
       "9            3.215190  \n",
       "10           3.480988  \n",
       "11          -0.547034  \n",
       "12          -0.330095  \n",
       "13           0.879680  \n",
       "14           2.054275  \n",
       "15          -0.513809  \n",
       "16          -1.549642  \n",
       "17          -0.861692  \n",
       "18           3.989133  \n",
       "19           3.967634  \n",
       "20          -0.263645  \n",
       "21          -0.304688  \n",
       "22          -0.291007  \n",
       "23          -0.676024  \n",
       "24          -0.973093  \n",
       "25          -0.922279  \n",
       "26          -0.109248  \n",
       "27           0.334401  \n",
       "28          -0.193287  \n",
       "29                NaN  \n",
       "...               ...  \n",
       "12523             NaN  \n",
       "12524             NaN  \n",
       "12525       -0.799152  \n",
       "12526        1.978053  \n",
       "12527             NaN  \n",
       "12528       -0.081886  \n",
       "12529             NaN  \n",
       "12530             NaN  \n",
       "12531       -3.156160  \n",
       "12532             NaN  \n",
       "12533             NaN  \n",
       "12534        0.574793  \n",
       "12535             NaN  \n",
       "12536             NaN  \n",
       "12537        1.329192  \n",
       "12538        1.260788  \n",
       "12539             NaN  \n",
       "12540             NaN  \n",
       "12541       -1.829121  \n",
       "12542             NaN  \n",
       "12543       -1.543779  \n",
       "12544        0.518115  \n",
       "12545             NaN  \n",
       "12546             NaN  \n",
       "12547             NaN  \n",
       "12548       -4.602418  \n",
       "12549        0.240590  \n",
       "12550             NaN  \n",
       "12551       -0.369183  \n",
       "12552             NaN  \n",
       "\n",
       "[12553 rows x 86 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ProtExpressDF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# change column name\n",
    "# helped: https://stackoverflow.com/questions/11346283/renaming-columns-in-pandas\n",
    "ProtExpressDF.rename(columns={'RefSeq_accession_number': 'RefSeq_accession'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RefSeq_accession</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>AO-A12D.01TCGA</th>\n",
       "      <th>C8-A131.01TCGA</th>\n",
       "      <th>AO-A12B.01TCGA</th>\n",
       "      <th>BH-A18Q.02TCGA</th>\n",
       "      <th>C8-A130.02TCGA</th>\n",
       "      <th>C8-A138.03TCGA</th>\n",
       "      <th>E2-A154.03TCGA</th>\n",
       "      <th>...</th>\n",
       "      <th>AO-A12B.34TCGA</th>\n",
       "      <th>A2-A0SW.35TCGA</th>\n",
       "      <th>AO-A0JL.35TCGA</th>\n",
       "      <th>BH-A0BV.35TCGA</th>\n",
       "      <th>A2-A0YM.36TCGA</th>\n",
       "      <th>BH-A0C7.36TCGA</th>\n",
       "      <th>A2-A0SX.36TCGA</th>\n",
       "      <th>263d3f-I.CPTAC</th>\n",
       "      <th>blcdb9-I.CPTAC</th>\n",
       "      <th>c4155b-C.CPTAC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_958782</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1</td>\n",
       "      <td>1.096131</td>\n",
       "      <td>2.609943</td>\n",
       "      <td>-0.659828</td>\n",
       "      <td>0.195341</td>\n",
       "      <td>-0.494060</td>\n",
       "      <td>2.765081</td>\n",
       "      <td>0.862659</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.963904</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.398560</td>\n",
       "      <td>0.598585</td>\n",
       "      <td>-0.191285</td>\n",
       "      <td>0.566975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_958785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1g</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.648742</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.938210</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.658143</td>\n",
       "      <td>-0.547749</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.606697</td>\n",
       "      <td>-0.183918</td>\n",
       "      <td>0.578702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NP_958786</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1a</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.654285</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.500619</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.943919</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows  86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  RefSeq_accession gene_symbol           gene_name  AO-A12D.01TCGA  \\\n",
       "0        NP_958782        PLEC   plectin isoform 1        1.096131   \n",
       "1        NP_958785         NaN  plectin isoform 1g        1.111370   \n",
       "2        NP_958786        PLEC  plectin isoform 1a        1.111370   \n",
       "\n",
       "   C8-A131.01TCGA  AO-A12B.01TCGA  BH-A18Q.02TCGA  C8-A130.02TCGA  \\\n",
       "0        2.609943       -0.659828        0.195341       -0.494060   \n",
       "1        2.650422       -0.648742        0.215413       -0.503899   \n",
       "2        2.650422       -0.654285        0.215413       -0.500619   \n",
       "\n",
       "   C8-A138.03TCGA  E2-A154.03TCGA       ...        AO-A12B.34TCGA  \\\n",
       "0        2.765081        0.862659       ...             -0.963904   \n",
       "1        2.779709        0.870186       ...             -0.938210   \n",
       "2        2.779709        0.870186       ...             -0.943919   \n",
       "\n",
       "   A2-A0SW.35TCGA  AO-A0JL.35TCGA  BH-A0BV.35TCGA  A2-A0YM.36TCGA  \\\n",
       "0       -0.487772        -0.10668       -0.065838        0.655850   \n",
       "1       -0.487772        -0.10668       -0.055893        0.658143   \n",
       "2       -0.487772        -0.10668       -0.065838        0.655850   \n",
       "\n",
       "   BH-A0C7.36TCGA  A2-A0SX.36TCGA  263d3f-I.CPTAC  blcdb9-I.CPTAC  \\\n",
       "0       -0.552212       -0.398560        0.598585       -0.191285   \n",
       "1       -0.547749       -0.392601        0.606697       -0.183918   \n",
       "2       -0.552212       -0.392601        0.603993       -0.186022   \n",
       "\n",
       "   c4155b-C.CPTAC  \n",
       "0        0.566975  \n",
       "1        0.578702  \n",
       "2        0.576747  \n",
       "\n",
       "[3 rows x 86 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ProtExpressDF.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RefSeqProteinID</th>\n",
       "      <th>Species</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIA</td>\n",
       "      <td>NP_006524</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>melanoma inhibitory activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_002002</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_998812</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol RefSeqProteinID       Species  \\\n",
       "0        MIA       NP_006524  Homo sapiens   \n",
       "1      FGFR4       NP_002002  Homo sapiens   \n",
       "2      FGFR4       NP_998812  Homo sapiens   \n",
       "\n",
       "                             Gene Name  \n",
       "0         melanoma inhibitory activity  \n",
       "1  fibroblast growth factor receptor 4  \n",
       "2  fibroblast growth factor receptor 4  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PAM50DF.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# change column name\n",
    "PAM50DF.rename(columns={'RefSeqProteinID': 'RefSeq_accession'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RefSeq_accession</th>\n",
       "      <th>Species</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIA</td>\n",
       "      <td>NP_006524</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>melanoma inhibitory activity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol RefSeq_accession       Species                     Gene Name\n",
       "0        MIA        NP_006524  Homo sapiens  melanoma inhibitory activity"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PAM50DF.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "100"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# count number of rows\n",
    "# helped: https://stackoverflow.com/questions/15943769/how-do-i-get-the-row-count-of-a-pandas-dataframe\n",
    "# note df.count()[0] will only return the count of non-NA/NaN rows\n",
    "PAM50DF.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# list3TRUEs = [TRUE]*3\n",
    "# must have quotes '' around TRUE for the list to generate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[True, True, True, True, True]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make a list with 100 'TRUE'\n",
    "# helped: https://stackoverflow.com/questions/3459098/create-list-of-single-item-repeated-n-times-in-python\n",
    "list5TRUEs = [True]*5\n",
    "list5TRUEs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "list100TRUEs = [True]*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    True\n",
       "1    True\n",
       "2    True\n",
       "3    True\n",
       "4    True\n",
       "dtype: bool"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make a series\n",
    "# helped: https://pandas.pydata.org/pandas-docs/stable/dsintro.html\n",
    "# note .Series must be capitalized!\n",
    "TRUEseries = pd.Series(list100TRUEs)\n",
    "TRUEseries.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RefSeq_accession</th>\n",
       "      <th>Species</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>PAM50included</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MIA</td>\n",
       "      <td>NP_006524</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>melanoma inhibitory activity</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_002002</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FGFR4</td>\n",
       "      <td>NP_998812</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>fibroblast growth factor receptor 4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol RefSeq_accession       Species  \\\n",
       "0        MIA        NP_006524  Homo sapiens   \n",
       "1      FGFR4        NP_002002  Homo sapiens   \n",
       "2      FGFR4        NP_998812  Homo sapiens   \n",
       "\n",
       "                             Gene Name  PAM50included  \n",
       "0         melanoma inhibitory activity           True  \n",
       "1  fibroblast growth factor receptor 4           True  \n",
       "2  fibroblast growth factor receptor 4           True  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# append series to PAM50DF with header = PAM50included \n",
    "PAM50DF['PAM50included'] = TRUEseries\n",
    "PAM50DF.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RefSeq_accession</th>\n",
       "      <th>Species</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>PAM50included</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>GRB7</td>\n",
       "      <td>NP_005301</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>growth factor receptor-bound protein 7</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>MELK</td>\n",
       "      <td>NP_055606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>maternal embryonic leucine zipper kinase</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>UBE2T</td>\n",
       "      <td>NP_054895</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ubiquitin-conjugating enzyme E2T (putative)</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneSymbol RefSeq_accession       Species  \\\n",
       "97       GRB7        NP_005301  Homo sapiens   \n",
       "98       MELK        NP_055606  Homo sapiens   \n",
       "99      UBE2T        NP_054895  Homo sapiens   \n",
       "\n",
       "                                      Gene Name  PAM50included  \n",
       "97       growth factor receptor-bound protein 7           True  \n",
       "98     maternal embryonic leucine zipper kinase           True  \n",
       "99  ubiquitin-conjugating enzyme E2T (putative)           True  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PAM50DF.tail(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
